### PR TITLE
feat: CTA component (with all states and MO button)

### PR DESCRIPTION
**Pull request by lilyfromseattle**
[PURCHASE-2702]

As a collector I want to see a CTA to make an offer, or displaying the specific status of an offer, when accessing a conversation about an artwork that fulfills the required conditions to be offer-able from inquiry.

Acceptance criteria:
-All states should be accounted for, and displayed given the conditions.
-No action should be accounted for (as of this ticket) when engaging with the CTA.

Because of the way. the components are configured, I also ended up adding the make offer button and hooking it up to the modal that Joel made. 

Screen Shot 2021-06-16 at 9 39 16 AM

Screen Shot 2021-06-16 at 9 38 37 AM

[PURCHASE-2702]: https://artsyproduct.atlassian.net/browse/PURCHASE-2702

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please use `<SystemQueryRenderer />` instead of `<QueryRender />`. This prevents double fetching during the server-side render pass. See:
> `src/v2/Apps/Conversation/Components/OpenInquiryModalButton.tsx`
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 46fb7eb07a2e2ade2f8f02294a3635e8635447f5
</p>

File: src/v2/Apps/Conversation/Components/OpenInquiryModalButton.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { OpenInquiryModalButtonQuery } from "v2/__generated__/OpenInquiryModalButtonQuery.graphql"
+import { Button, CheckCircleIcon, Flex, Separator, Text } from "@artsy/palette"
+import React from "react"
+import styled from "styled-components"
+import { graphql, QueryRenderer } from "react-relay"
```
Comment (damassi) [inline src/v2/Apps/Conversation/Components/OpenInquiryModalButton.tsx:5]: (See the danger error that was added as a PR comment)

File: src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  const navigateToConversation = (path: string, title: string) => {
+    trackEvent(
+      tappedViewOffer({
+        impulse_conversation_id: conversationID,
+        cta: message,
+      })
+    )
+    window.location.assign(path)
+    // figure out how to pass props { orderID, title }
```
Comment (damassi) [inline src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx:146]: You should be able to use our router to transition to the conversation more smoothly (in a single page app kind of way, without the hard jump between pages); and to pass `orderID` and `title`, can use queryParams on the url, which you can then pluck off later:  

```tsx
const { router } = useRouter()
... 
router.push(`${path}?orderID=foo&title=bar`)
```

**Review by damassi — COMMENTED** [review]
Really nice work @lilyfromseattle, just a couple minor comments 👍

File: src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx (hunk 2 above)
Comment (lilyfromseattle) [reply src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx]: Ahhh right, I forgot what the right way to do this was so I went googling

**Pull request merged**